### PR TITLE
Major cleanup of publisher sources

### PR DESCRIPTION
**Pull request by barnstee**
Using statements cleanup
Model definition file cleanup
Leveraging UA cert for username/password crypto instead of IoT Edge cert in Edge context only
Removed legacy connection string hiding in cert
Removed unused interfaces
Renamed classes to better reflect their use
Fixed singleton pattern

### CODE REVIEW

**Comment by koepalex** [comment]
> using System;

Copyright header is missing

---
Refers to: opcpublisher/EncryptedNetworkCredential.cs:1 in 9c13d94. [](commit_id = 9c13d94ecc06a8020a6908743951a0007bc62209, deletion_comment = False)

**Comment by koepalex** [comment]
> using Microsoft.Azure.Devices.Client;

Copyright header is missing

---
Refers to: opcpublisher/HubClient.cs:1 in 9c13d94. [](commit_id = 9c13d94ecc06a8020a6908743951a0007bc62209, deletion_comment = False)

**Comment by koepalex** [comment]
> using Microsoft.Azure.Devices.Client;

namespaces should be either outside of namespace or within, in HubClient.cs was refactoring to have usings outside

---
Refers to: opcpublisher/HubCommunicationBase.cs:1 in 9c13d94. [](commit_id = 9c13d94ecc06a8020a6908743951a0007bc62209, deletion_comment = False)

**Comment by koepalex** [comment]
> using Microsoft.Azure.Devices.Client;

Copyright header is missing

---
Refers to: opcpublisher/HubCommunicationBase.cs:1 in 9c13d94. [](commit_id = 9c13d94ecc06a8020a6908743951a0007bc62209, deletion_comment = False)

**Comment by koepalex** [comment]
> using System;

Copyright header is missing #Resolved

---
Refers to: opcpublisher/IotEdgeHubCommunication.cs:1 in 9c13d94. [](commit_id = 9c13d94ecc06a8020a6908743951a0007bc62209, deletion_comment = False)

**Comment by koepalex** [comment]
> using Opc.Ua;

Copyright header is missing #Resolved

---
Refers to: opcpublisher/OpcApplicationConfiguration.cs:1 in 9c13d94. [](commit_id = 9c13d94ecc06a8020a6908743951a0007bc62209, deletion_comment = False)

**Comment by koepalex** [comment]
> using Opc.Ua;

Copyright header is missing #Resolved

---
Refers to: opcpublisher/OpcUaMonitoredItem.cs:1 in 9c13d94. [](commit_id = 9c13d94ecc06a8020a6908743951a0007bc62209, deletion_comment = False)

**Comment by koepalex** [comment]
> using System.Collections.Generic;

Copyright header is missing

---
Refers to: opcpublisher/Models/UnpublishNodesMethodRequestModel.cs:1 in 9c13d94. [](commit_id = 9c13d94ecc06a8020a6908743951a0007bc62209, deletion_comment = False)

**Comment by koepalex** [comment]
> namespace OpcPublisher

   Copyright header is missing

---
Refers to: opcpublisher/Models/UnpublishAllNodesMethodRequestModel.cs:1 in 9c13d94. [](commit_id = 9c13d94ecc06a8020a6908743951a0007bc62209, deletion_comment = False)

File: opcpublisher/Models/UnpublishNodesMethodRequestModel.cs [hunk 1]
```diff
@@ -0,0 +1,20 @@
+﻿using System.Collections.Generic;
+
+namespace OpcPublisher
```
Comment (koepalex) [inline opcpublisher/Models/UnpublishNodesMethodRequestModel.cs:8]: >OpcPublisher [](start = 10, length = 12)

namespaces dosn't match sub-folder structure

**Comment by koepalex** [comment]
> namespace OpcPublisher

namespaces dosn't match sub-folder structure

---
Refers to: opcpublisher/Models/UnpublishAllNodesMethodRequestModel.cs:1 in 9c13d94. [](commit_id = 9c13d94ecc06a8020a6908743951a0007bc62209, deletion_comment = False)

File: opcpublisher/Models/PublishNodesMethodRequestModel.cs [hunk 2]
```diff
@@ -0,0 +1,37 @@
+﻿using Newtonsoft.Json;
+using Newtonsoft.Json.Converters;
+using System.Collections.Generic;
+
+namespace OpcPublisher
```
Comment (koepalex) [inline opcpublisher/Models/PublishNodesMethodRequestModel.cs:10]: >OpcPublisher [](start = 10, length = 12)

namespaces dosn't match sub-folder structure

**Comment by koepalex** [comment]
> using Newtonsoft.Json;

Copyright header is missing

---
Refers to: opcpublisher/Models/PublishNodesMethodRequestModel.cs:1 in 9c13d94. [](commit_id = 9c13d94ecc06a8020a6908743951a0007bc62209, deletion_comment = False)

**Comment by koepalex** [comment]
> using Newtonsoft.Json;

Copyright header is missing

---
Refers to: opcpublisher/Models/OpcNodeOnEndpointModel.cs:1 in 9c13d94. [](commit_id = 9c13d94ecc06a8020a6908743951a0007bc62209, deletion_comment = False)

File: opcpublisher/Models/OpcNodeOnEndpointModel.cs [hunk 3]
```diff
@@ -0,0 +1,46 @@
+﻿using Newtonsoft.Json;
+
+namespace OpcPublisher
```
Comment (koepalex) [inline opcpublisher/Models/OpcNodeOnEndpointModel.cs:8]: namespaces dosn't match sub-folder structure

**Comment by koepalex** [comment]
> using Opc.Ua;

Copyright header is missing

---
Refers to: opcpublisher/Models/NodeConfigurationModel.cs:1 in 9c13d94. [](commit_id = 9c13d94ecc06a8020a6908743951a0007bc62209, deletion_comment = False)

File: opcpublisher/Models/NodeConfigurationModel.cs [hunk 4]
```diff
@@ -0,0 +1,112 @@
+﻿using Opc.Ua;
+using OpcPublisher.Crypto;
+
+namespace OpcPublisher
```
Comment (koepalex) [inline opcpublisher/Models/NodeConfigurationModel.cs:8]: namespaces dosn't match sub-folder structure

**Comment by koepalex** [comment]
> using Newtonsoft.Json;

Copyright header is missing

---
Refers to: opcpublisher/Models/GetInfoMethodResponseModel.cs:1 in 9c13d94. [](commit_id = 9c13d94ecc06a8020a6908743951a0007bc62209, deletion_comment = False)

File: opcpublisher/Models/GetInfoMethodResponseModel.cs [hunk 5]
```diff
@@ -0,0 +1,39 @@
+﻿using Newtonsoft.Json;
+using System.Runtime.InteropServices;
+
+namespace OpcPublisher
```
Comment (koepalex) [inline opcpublisher/Models/GetInfoMethodResponseModel.cs:9]: namespaces dosn't match sub-folder structure

**Comment by koepalex** [comment]
> using Newtonsoft.Json;

Copyright header is missing

---
Refers to: opcpublisher/Models/GetConfiguredNodesOnEndpointMethodResponseModel.cs:1 in 9c13d94. [](commit_id = 9c13d94ecc06a8020a6908743951a0007bc62209, deletion_comment = False)

File: opcpublisher/Models/GetConfiguredNodesOnEndpointMethodResponseModel.cs [hunk 6]
```diff
@@ -0,0 +1,31 @@
+﻿using Newtonsoft.Json;
+using System.Collections.Generic;
+
+namespace OpcPublisher
```
Comment (koepalex) [inline opcpublisher/Models/GetConfiguredNodesOnEndpointMethodResponseModel.cs:9]: namespaces dosn't match sub-folder structure

**Comment by koepalex** [comment]
> using Newtonsoft.Json;

Copyright header is missing

---
Refers to: opcpublisher/Models/GetConfiguredNodesOnEndpointMethodRequestModel.cs:1 in 9c13d94. [](commit_id = 9c13d94ecc06a8020a6908743951a0007bc62209, deletion_comment = False)

File: opcpublisher/Models/GetConfiguredNodesOnEndpointMethodRequestModel.cs [hunk 7]
```diff
@@ -0,0 +1,21 @@
+﻿using Newtonsoft.Json;
+
+namespace OpcPublisher
```
Comment (koepalex) [inline opcpublisher/Models/GetConfiguredNodesOnEndpointMethodRequestModel.cs:8]: namespaces dosn't match sub-folder structure

**Comment by koepalex** [comment]
> using Newtonsoft.Json;

Copyright header is missing

---
Refers to: opcpublisher/Models/GetConfiguredEndpointsMethodResponseModel.cs:1 in 9c13d94. [](commit_id = 9c13d94ecc06a8020a6908743951a0007bc62209, deletion_comment = False)

File: opcpublisher/Models/GetConfiguredEndpointsMethodResponseModel.cs [hunk 8]
```diff
@@ -0,0 +1,25 @@
+﻿using Newtonsoft.Json;
+using System.Collections.Generic;
+
+namespace OpcPublisher
```
Comment (koepalex) [inline opcpublisher/Models/GetConfiguredEndpointsMethodResponseModel.cs:9]: namespaces dosn't match sub-folder structure

**Comment by koepalex** [comment]
> using Newtonsoft.Json;

Copyright header is missing

---
Refers to: opcpublisher/Models/GetConfiguredEndpointsMethodRequestModel.cs:1 in 9c13d94. [](commit_id = 9c13d94ecc06a8020a6908743951a0007bc62209, deletion_comment = False)

File: opcpublisher/Models/GetConfiguredEndpointsMethodRequestModel.cs [hunk 9]
```diff
@@ -0,0 +1,18 @@
+﻿using Newtonsoft.Json;
+
+namespace OpcPublisher
```
Comment (koepalex) [inline opcpublisher/Models/GetConfiguredEndpointsMethodRequestModel.cs:8]: namespaces dosn't match sub-folder structure

**Comment by koepalex** [comment]
> using Newtonsoft.Json;

Copyright header is missing

---
Refers to: opcpublisher/Models/ExitApplicationMethodRequestModel.cs:1 in 9c13d94. [](commit_id = 9c13d94ecc06a8020a6908743951a0007bc62209, deletion_comment = False)

File: opcpublisher/Models/ExitApplicationMethodRequestModel.cs [hunk 10]
```diff
@@ -0,0 +1,17 @@
+﻿using Newtonsoft.Json;
+
+namespace OpcPublisher
```
Comment (koepalex) [inline opcpublisher/Models/ExitApplicationMethodRequestModel.cs:8]: namespaces dosn't match sub-folder structure

**Comment by koepalex** [comment]
> using Newtonsoft.Json;

Copyright header is missing

---
Refers to: opcpublisher/Models/DiagnosticLogMethodResponseModel.cs:1 in 9c13d94. [](commit_id = 9c13d94ecc06a8020a6908743951a0007bc62209, deletion_comment = False)

File: opcpublisher/Models/DiagnosticLogMethodResponseModel.cs [hunk 11]
```diff
@@ -0,0 +1,24 @@
+﻿using Newtonsoft.Json;
+using System.Collections.Generic;
+
+namespace OpcPublisher
```
Comment (koepalex) [inline opcpublisher/Models/DiagnosticLogMethodResponseModel.cs:9]: namespaces dosn't match sub-folder structure

**Comment by koepalex** [comment]
> using Microsoft.Azure.Devices.Client;

Copyright header is missing

---
Refers to: opcpublisher/Models/DiagnosticInfoMethodResponseModel.cs:1 in 9c13d94. [](commit_id = 9c13d94ecc06a8020a6908743951a0007bc62209, deletion_comment = False)

File: opcpublisher/Models/DiagnosticInfoMethodResponseModel.cs [hunk 12]
```diff
@@ -0,0 +1,85 @@
+﻿using Microsoft.Azure.Devices.Client;
+using Newtonsoft.Json;
+using System;
+
+namespace OpcPublisher
```
Comment (koepalex) [inline opcpublisher/Models/DiagnosticInfoMethodResponseModel.cs:10]: namespaces dosn't match sub-folder structure

**Comment by koepalex** [comment]
> namespace OpcPublisher

Copyright header is missing

---
Refers to: opcpublisher/Models/ConfiguredEndpointModel.cs:1 in 9c13d94. [](commit_id = 9c13d94ecc06a8020a6908743951a0007bc62209, deletion_comment = False)

**Comment by koepalex** [comment]
> namespace OpcPublisher

namespaces dosn't match sub-folder structure

---
Refers to: opcpublisher/Models/ConfiguredEndpointModel.cs:1 in 9c13d94. [](commit_id = 9c13d94ecc06a8020a6908743951a0007bc62209, deletion_comment = False)

**Comment by koepalex** [comment]
> using Newtonsoft.Json;

Copyright header is missing

---
Refers to: opcpublisher/Models/ConfigurationFileEntryModel.cs:1 in 9c13d94. [](commit_id = 9c13d94ecc06a8020a6908743951a0007bc62209, deletion_comment = False)

File: opcpublisher/Models/ConfigurationFileEntryModel.cs [hunk 13]
```diff
@@ -0,0 +1,88 @@
+﻿using Newtonsoft.Json;
+using Newtonsoft.Json.Converters;
+using OpcPublisher.Crypto;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+
+namespace OpcPublisher
```
Comment (koepalex) [inline opcpublisher/Models/ConfigurationFileEntryModel.cs:12]: namespaces dosn't match sub-folder structure

**Comment by koepalex** [comment]
> using Newtonsoft.Json;

Copyright header is missing

---
Refers to: opcpublisher/Models/ConfigurationFileEntryLegacyModel.cs:1 in 9c13d94. [](commit_id = 9c13d94ecc06a8020a6908743951a0007bc62209, deletion_comment = False)

File: opcpublisher/Models/ConfigurationFileEntryLegacyModel.cs [hunk 14]
```diff
@@ -0,0 +1,112 @@
+﻿using Newtonsoft.Json;
+using Newtonsoft.Json.Converters;
+using Opc.Ua;
+using OpcPublisher.Crypto;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+
+namespace OpcPublisher
```
Comment (koepalex) [inline opcpublisher/Models/ConfigurationFileEntryLegacyModel.cs:13]: namespaces dosn't match sub-folder structure

**Comment by koepalex** [comment]
> using System.Collections.Generic;

Copyright header is missing

---
Refers to: opcpublisher/Interfaces/IPublisherNodeConfiguration.cs:1 in 9c13d94. [](commit_id = 9c13d94ecc06a8020a6908743951a0007bc62209, deletion_comment = False)

File: opcpublisher/Interfaces/IPublisherNodeConfiguration.cs [hunk 15]
```diff
@@ -1,10 +1,9 @@
 ﻿using System.Collections.Generic;
+using System.Threading;
 using System.Threading.Tasks;
 
 namespace OpcPublisher
```
Comment (koepalex) [inline opcpublisher/Interfaces/IPublisherNodeConfiguration.cs]: namespaces dosn't match sub-folder structure, should be OpcPublisher.Interfaces

File: opcpublisher/Interfaces/IPublisherNodeConfiguration.cs [hunk 16]
```diff
@@ -1,10 +1,9 @@
...
 {
-    using System.Threading;
-
     public interface IPublisherNodeConfiguration
```
Comment (koepalex) [inline opcpublisher/Interfaces/IPublisherNodeConfiguration.cs:12]: access modifier can be changed to internal (maybe in combination with InternalsVisibleTo attribute)

**Comment by koepalex** [comment]
>     public interface IHubCommunication

access modifier can be changed to internal (maybe in combination with InternalsVisibleTo attribute)

---
Refers to: opcpublisher/Interfaces/IHubCommunication.cs:11 in 9c13d94. [](commit_id = 9c13d94ecc06a8020a6908743951a0007bc62209, deletion_comment = False)

File: opcpublisher/Interfaces/IHubCommunication.cs [hunk 17]
```diff
@@ -1,11 +1,10 @@
-﻿using System.Threading;
+﻿using Microsoft.Azure.Devices.Client;
+using System.Collections.Generic;
+using System.Threading;
 using System.Threading.Tasks;
 
 namespace OpcPublisher
```
Comment (koepalex) [inline opcpublisher/Interfaces/IHubCommunication.cs]: namespaces dosn't match sub-folder structure, should be OpcPublisher.Interfaces

**Comment by koepalex** [comment]
> using Microsoft.Azure.Devices.Client;

Copyright header is missing

---
Refers to: opcpublisher/Interfaces/IHubCommunication.cs:1 in 9c13d94. [](commit_id = 9c13d94ecc06a8020a6908743951a0007bc62209, deletion_comment = False)

**Comment by koepalex** [comment]
> using Microsoft.Azure.Devices.Client;

Copyright header is missing

---
Refers to: opcpublisher/Interfaces/IHubClient.cs:1 in 9c13d94. [](commit_id = 9c13d94ecc06a8020a6908743951a0007bc62209, deletion_comment = False)

File: opcpublisher/Interfaces/IHubClient.cs [hunk 18]
```diff
@@ -1,9 +1,8 @@
-﻿using System.Threading.Tasks;
+﻿using Microsoft.Azure.Devices.Client;
+using System.Threading.Tasks;
 
 namespace OpcPublisher
```
Comment (koepalex) [inline opcpublisher/Interfaces/IHubClient.cs]: namespaces dosn't match sub-folder structure, should be OpcPublisher.Interfaces

**Comment by koepalex** [comment]
>     public interface IHubClient

access modifier can be changed to internal (maybe in combination with InternalsVisibleTo attribute)

---
Refers to: opcpublisher/Interfaces/IHubClient.cs:9 in 9c13d94. [](commit_id = 9c13d94ecc06a8020a6908743951a0007bc62209, deletion_comment = False)

File: opcpublisher/PublisherTelemetryConfiguration.cs [hunk 19]
```diff
@@ -393,7 +392,7 @@ public TelemetryConfigurationFileModel()
         }
     }
 
-    public class PublisherTelemetryConfiguration : IPublisherTelemetryConfiguration
+    public class PublisherTelemetryConfiguration
```
Comment (koepalex) [inline opcpublisher/PublisherTelemetryConfiguration.cs:16]: extract in own file?

**Comment by koepalex** [comment]
> using Newtonsoft.Json;

Copyright header is missing

---
Refers to: opcpublisher/PublisherTelemetryConfiguration.cs:1 in 9c13d94. [](commit_id = 9c13d94ecc06a8020a6908743951a0007bc62209, deletion_comment = False)

**Comment by koepalex** [comment]
> using Opc.Ua;

Copyright header is missing

---
Refers to: opcpublisher/PublisherServer.cs:1 in 9c13d94. [](commit_id = 9c13d94ecc06a8020a6908743951a0007bc62209, deletion_comment = False)

File: opcpublisher/PublisherNodeManager.cs [hunk 20]
```diff
@@ -1,22 +1,21 @@
+using Microsoft.Azure.Devices.Client;
```
Comment (koepalex) [inline opcpublisher/PublisherNodeManager.cs:6]: Copyright header is missing

**Comment by koepalex** [comment]
> using Serilog;

Copyright header is missing

---
Refers to: opcpublisher/PublisherDiagnostics.cs:1 in 9c13d94. [](commit_id = 9c13d94ecc06a8020a6908743951a0007bc62209, deletion_comment = False)

File: opcpublisher/PublisherDiagnostics.cs [hunk 21]
```diff
@@ -124,7 +130,7 @@ public DiagnosticInfoMethodResponseModel GetDiagnosticInfo()
             catch (Exception ex)
             {
                 // startup might be not completed yet
-                Logger.Error(ex, "Collecting diagnostics information causing error {diagnosticInfo}", diagnosticInfo);
+                Program.Logger.Error(ex, "Collecting diagnostics information causing error {diagnosticInfo}", diagnosticInfo);
```
Comment (koepalex) [inline opcpublisher/PublisherDiagnostics.cs]: Why do we exchange one static instance (Logger) with Program.Logger? There are the equal, I would prefer passing ILogger interface into class and using member

**Comment by koepalex** [comment]
>     public class DiagnosticLogSink : ILogEventSink

Extract in own file?

---
Refers to: opcpublisher/PublisherDiagnostics.cs:327 in 9c13d94. [](commit_id = 9c13d94ecc06a8020a6908743951a0007bc62209, deletion_comment = False)

**Comment by koepalex** [comment]
> using Mono.Options;

Copyright header is missing

---
Refers to: opcpublisher/Program.cs:1 in 9c13d94. [](commit_id = 9c13d94ecc06a8020a6908743951a0007bc62209, deletion_comment = False)

**Comment by koepalex** [comment]
> using Opc.Ua.Client;

Copyright header is missing

---
Refers to: opcpublisher/OpcUaSubscriptionWrapper.cs:1 in 9c13d94. [](commit_id = 9c13d94ecc06a8020a6908743951a0007bc62209, deletion_comment = False)

**Comment by koepalex** [comment]
> using System;

Copyright header is missing

---
Refers to: opcpublisher/OpcUaSubscriptionManager.cs:1 in 9c13d94. [](commit_id = 9c13d94ecc06a8020a6908743951a0007bc62209, deletion_comment = False)

**Comment by koepalex** [comment]
> using Opc.Ua;

Copyright header is missing #Resolved

---
Refers to: opcpublisher/OpcUaSession.cs:1 in 9c13d94. [](commit_id = 9c13d94ecc06a8020a6908743951a0007bc62209, deletion_comment = False)

**Comment by koepalex** [comment]
> using Opc.Ua;

Copyright header is missing #Closed

---
Refers to: opcpublisher/OpcSession.cs:1 in 9c13d94. [](commit_id = 9c13d94ecc06a8020a6908743951a0007bc62209, deletion_comment = False)

**Comment by koepalex** [comment]
> using Opc.Ua;

Copyright header is missing #Resolved

---
Refers to: opcpublisher/OpcMonitoredItem.cs:1 in 9c13d94. [](commit_id = 9c13d94ecc06a8020a6908743951a0007bc62209, deletion_comment = False)

**Comment by koepalex** [comment]
> using static OpcPublisher.Program;

Copyright header is missing

---
Refers to: opcpublisher/MessageData.cs:1 in 9c13d94. [](commit_id = 9c13d94ecc06a8020a6908743951a0007bc62209, deletion_comment = False)

File: opcpublisher/MessageData.cs [hunk 22]
```diff
@@ -0,0 +1,119 @@
+﻿using static OpcPublisher.Program;
+
+namespace OpcPublisher
+{
+    /// <summary>
+    /// Class used to pass data from the MonitoredItem notification to the hub message processing.
+    /// </summary>
+    public class MessageData
```
Comment (koepalex) [inline opcpublisher/MessageData.cs]: move file to models?

**Comment by koepalex** [comment]
> using Microsoft.Azure.Devices;

Copyright header is missing

---
Refers to: opcpublisher/IotHubCommunication.cs:1 in 9c13d94. [](commit_id = 9c13d94ecc06a8020a6908743951a0007bc62209, deletion_comment = False)

File: opcpublisher/EncryptedNetworkCredential.cs [hunk 23]
```diff
@@ -0,0 +1,67 @@
+﻿using System;
+using System.Net;
+using System.Security.Cryptography.X509Certificates;
+using System.Threading.Tasks;
+
+namespace OpcPublisher.Crypto
```
Comment (koepalex) [inline opcpublisher/EncryptedNetworkCredential.cs]: >OpcPublisher.Crypto [](start = 10, length = 19)

Please move file into folder "Crypto"

**Comment by koepalex** [comment]
> using Newtonsoft.Json;

Copyright header is missing

---
Refers to: opcpublisher/PublisherNodeConfiguration.cs:1 in 9c13d94. [](commit_id = 9c13d94ecc06a8020a6908743951a0007bc62209, deletion_comment = False)

File: opcpublisher/EncryptedNetworkCredential.cs (hunk 23 above)
Comment (barnstee) [reply opcpublisher/EncryptedNetworkCredential.cs]: crypto folder was removed as it only had one file in it.

**Review by mregen — APPROVED** [review]
:shipit:

File: opcpublisher/Interfaces/IPublisherNodeConfiguration.cs (hunk 16 above)
Comment (barnstee) [reply opcpublisher/Interfaces/IPublisherNodeConfiguration.cs:12]: Nope, it breaks unit tests.

**Review by koepalex — APPROVED** [review]
:shipit:

**Pull request merged**